resteasy-cdi/src/main/java/org/jboss/resteasy/cdi/ResteasyCdiExtension.java

```diff
@@ -271,7 +272,26 @@ private void setBeanManager(BeanManager beanManager)
     */
    private boolean isFinalClass(Class clazz) {
       // Unproxyable bean type: classes which are declared final
-      return Modifier.isFinal(clazz.getModifiers());
-   }
+      boolean isFinal = Modifier.isFinal(clazz.getModifiers());
 
+      if (!isFinal) {
+         // check methods
+         for (Method m : clazz.getMethods()) {
+            if (clazz == m.getDeclaringClass())
+            {
+               int mod = m.getModifiers();
+               if (Modifier.isFinal(mod) && !Modifier.isStatic(mod)
+                  && (Modifier.isPublic(mod) || Modifier.isProtected(mod)))
```

Comment:
I think !Modifier.isPrivate(mod) would be better: it would capture the default visibility case as well.

Reply:
@rsearls, great. Thanks.

I don't know why, but github says you made the change "3 hours from now". We should contact some quantum physicists. Lol.